packages/react-native-reanimated/src/component/PerformanceMonitor.tsx

```diff
@@ -141,31 +122,24 @@ function JsPerformance() {
   );
 }
 
-function UiPerformance() {
+function UiPerformance({ smoothingFrames }: { smoothingFrames: number }) {
   const uiFps = useSharedValue<string | null>(null);
-  const totalRenderTime = useSharedValue(0);
   const circularBuffer = useSharedValue<CircularBuffer | null>(null);
 
   useFrameCallback(({ timestamp }: FrameInfo) => {
     if (circularBuffer.value === null) {
-      circularBuffer.value = createCircularDoublesBuffer(DEFAULT_BUFFER_SIZE);
+      circularBuffer.value = createCircularDoublesBuffer(smoothingFrames);
     }
 
     timestamp = Math.round(timestamp);
-    const previousTimestamp = circularBuffer.value.front() ?? timestamp;
 
-    const currentFps = completeBufferRoutine(
-      circularBuffer.value,
-      timestamp,
-      previousTimestamp,
-      totalRenderTime
-    );
+    const currentFps = completeBufferRoutine(circularBuffer.value, timestamp);
 
     uiFps.value = currentFps.toFixed(0);
   });
 
   const animatedProps = useAnimatedProps(() => {
-    const text = 'UI: ' + uiFps.value ?? 'N/A';
+    const text = 'UI: ' + (uiFps.value ?? 'N/A') + ' ';
```

Comment:
@bartlomiejbloniarz discovered some device specific glitches, where the FPS value was jumping or being cut-off.
Adding the space gave the text field more room and fixed the issue on his side.
Since I couldn't replicate this issue on my side, I decided to apply bb's patch instead of guessing which CSS rules would work best.

Reply:
alright, fine by me